### PR TITLE
feat: report page tsunami

### PR DESCRIPTION
**Pull request by whes1015**
## 這是什麼類型的 PR？

> 選擇所有符合的項目

- [ ] 重構
- [x] 新功能
- [ ] 錯誤修復
- [ ] 最佳化
- [ ] 技術文件更新

## 相關 issue

- 相關問題 #
- closes #

## QA 指南、截圖、錄像

> _請將這行替換成：如何測試您的 PR 的步驟，已測試的裝置註釋，以及任何相關的 UI 更改圖片。_

### UI 無障礙清單

> _如果您的 PR 包含 UI 更改，請使用此清單：_

- [ ] 變數名稱實現語意化命名？
- [ ] 測試通過 AA 顏色對比？

### CODE REVIEW

**Comment by kamiya10** [comment]
截圖

**Review by lowrt — APPROVED** [review]

File: lib/route/report/report.dart [hunk 1]
```diff
@@ -266,6 +266,38 @@ class _ReportRouteState extends State<ReportRoute> with TickerProviderStateMixin
...
+                          backgroundColor: Colors.red.withOpacity(0.16),
+                          labelStyle: const TextStyle(color: Colors.white, fontWeight: FontWeight.w900),
+                          side: const BorderSide(color: Colors.red),
+                          onPressed: () {},
+                        ),
+                      ),
+                    ],
+                  ),
+                )
+              : const SizedBox(),
```
Comment (kamiya10) [inline lib/route/report/report.dart:300]:
```suggestion
          if (report != null)
              Positioned(
                  right: 4,
                  top: context.padding.top + 4,
                  child: Row(
                    children: [
                      Visibility(
                        visible: report!.magnitude >= 6 && report!.magnitude < 7 && report!.getLocation().contains("海"),
                        child: ActionChip(
                          avatar: const Icon(Symbols.tsunami_rounded),
                          label: const Text("可能引起若干海面變動"),
                          backgroundColor: Colors.blue.withOpacity(0.16),
                          labelStyle: const TextStyle(fontWeight: FontWeight.w900),
                          side: const BorderSide(color: Colors.blue),
                          onPressed: () {},
                        ),
                      ),
                      Visibility(
                        visible: report!.magnitude >= 7 && report!.getLocation().contains("海"),
                        child: ActionChip(
                          avatar: const Icon(Symbols.tsunami_rounded, color: Colors.white),
                          label: const Text("可能引起海嘯 應注意後續資訊"),
                          backgroundColor: Colors.red.withOpacity(0.16),
                          labelStyle: const TextStyle(color: Colors.white, fontWeight: FontWeight.w900),
                          side: const BorderSide(color: Colors.red),
                          onPressed: () {},
                        ),
                      ),
                    ],
                  ),
                ),
```
Just use an if statement

File: lib/route/report/report.dart [hunk 2]
```diff
@@ -266,6 +266,38 @@ class _ReportRouteState extends State<ReportRoute> with TickerProviderStateMixin
...
+                        visible: report!.magnitude >= 7 && report!.getLocation().contains("海"),
+                        child: ActionChip(
+                          avatar: const Icon(Symbols.tsunami_rounded, color: Colors.white),
+                          label: const Text("可能引起海嘯 應注意後續資訊"),
+                          backgroundColor: Colors.red.withOpacity(0.16),
+                          labelStyle: const TextStyle(color: Colors.white, fontWeight: FontWeight.w900),
+                          side: const BorderSide(color: Colors.red),
+                          onPressed: () {},
+                        ),
+                      ),
```
Comment (kamiya10) [inline lib/route/report/report.dart:296]: Don't list out all possible widgets in a tree.  
Convert these `Visibility()`'s to if...else statement will be better.

File: lib/route/report/report.dart [hunk 3]
```diff
@@ -266,6 +266,38 @@ class _ReportRouteState extends State<ReportRoute> with TickerProviderStateMixin
...
           ),
+          (report != null)
+              ? Positioned(
+                  right: 4,
+                  top: context.padding.top + 4,
+                  child: Row(
+                    children: [
+                      Visibility(
+                        visible: report!.magnitude >= 6 && report!.magnitude < 7 && report!.getLocation().contains("海"),
+                        child: ActionChip(
```
Comment (kamiya10) [inline lib/route/report/report.dart:277]: Don't use `ActionChip()`'s for display ppurpose, see [Material 3's Guidelines on Chips](https://m3.material.io/components/chips/guidelines) for a clearer reference on how  and when to use it.

File: lib/route/report/report.dart [hunk 4]
```diff
@@ -266,6 +266,38 @@ class _ReportRouteState extends State<ReportRoute> with TickerProviderStateMixin
...
+                          label: const Text("可能引起若干海面變動"),
+                          backgroundColor: Colors.blue.withOpacity(0.16),
+                          labelStyle: const TextStyle(fontWeight: FontWeight.w900),
+                          side: const BorderSide(color: Colors.blue),
+                          onPressed: () {},
+                        ),
+                      ),
+                      Visibility(
+                        visible: report!.magnitude >= 7 && report!.getLocation().contains("海"),
+                        child: ActionChip(
```
Comment (kamiya10) [inline lib/route/report/report.dart:288]: Don't use `ActionChip()`'s for display ppurpose, see [Material 3's Guidelines on Chips](https://m3.material.io/components/chips/guidelines) for a clearer reference on how  and when to use it.

**Review by kamiya10 — CHANGES_REQUESTED** [review]

**Comment by kamiya10** [comment]
And please provide informative description on what you have changed and screenshots in pull requests.  
PRs with missing required fields will be anot merged and closed.

**Comment by whes1015** [comment]
image

File: lib/route/report/report.dart [hunk 5]
```diff
@@ -266,6 +267,42 @@ class _ReportRouteState extends State<ReportRoute> with TickerProviderStateMixin
               refreshReport();
             },
           ),
+          Positioned(
```
Comment (kamiya10) [inline lib/route/report/report.dart:270]:
```suggestion
          if (report != null)
              Positioned(
```

File: lib/route/report/report.dart [hunk 6]
```diff
@@ -266,6 +267,42 @@ class _ReportRouteState extends State<ReportRoute> with TickerProviderStateMixin
...
+            top: context.padding.top + 50,
+            left: 0,
+            right: 0,
+            child: Column(
+              children: [
+                if (report != null &&
+                    report!.magnitude >= 6 &&
```
Comment (kamiya10) [inline lib/route/report/report.dart:277]:
```suggestion
                if (report!.magnitude >= 6 &&
```

File: lib/route/report/report.dart [hunk 7]
```diff
@@ -266,6 +267,42 @@ class _ReportRouteState extends State<ReportRoute> with TickerProviderStateMixin
...
+                      "此地震可能引起若干海面變動",
+                      style: TextStyle(
+                        color: context.theme.extendedColors.blue,
+                      ),
+                    ),
+                    backgroundColor: Colors.blue.withOpacity(0.16),
+                    labelStyle: const TextStyle(fontWeight: FontWeight.w900),
+                    side: BorderSide(color: context.theme.extendedColors.blue),
+                  ),
+                if (report != null && report!.magnitude >= 7 && report!.getLocation().contains("海"))
```
Comment (kamiya10) [inline lib/route/report/report.dart:295]:
```suggestion
                if (report!.magnitude >= 7 && report!.getLocation().contains("海"))
```

**Review by kamiya10 — CHANGES_REQUESTED** [review]
some code improvements

**Review by yayacat — DISMISSED** [review]
@whes1015 pls fix some code improvements for @kamiya10

**Review by kamiya10 — APPROVED** [review]

**Pull request merged**